owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/RemoveFileUseCase.kt

```diff
@@ -23,17 +25,35 @@ import com.owncloud.android.domain.files.FileRepository
 import com.owncloud.android.domain.files.model.OCFile
 
 class RemoveFileUseCase(
-    private val fileRepository: FileRepository
+    private val fileRepository: FileRepository,
+    private val setLastUsageFileUseCase: SetLastUsageFileUseCase,
 ) : BaseUseCaseWithResult<Unit, RemoveFileUseCase.Params>() {
 
     override fun run(params: Params) {
 
         require(params.listOfFilesToDelete.isNotEmpty())
 
-        return fileRepository.deleteFiles(
+        val listOfFilesToDeleteOriginal = params.listOfFilesToDelete.map { it to it.isAvailableLocally }
+
+        val deleteFiles = fileRepository.deleteFiles(
             listOfFilesToDelete = params.listOfFilesToDelete,
-            removeOnlyLocalCopy = params.removeOnlyLocalCopy
+            removeOnlyLocalCopy = params.removeOnlyLocalCopy,
         )
+
+        if (params.removeOnlyLocalCopy) {
+            listOfFilesToDeleteOriginal.forEach { (ocFile, isAvailableLocally) ->
+                setLastUsageFile(ocFile, isAvailableLocally)
+            }
+        }
+        return deleteFiles
+    }
+
+    private fun setLastUsageFile(file: OCFile, isAvailableLocally: Boolean ){
+            setLastUsageFileUseCase(SetLastUsageFileUseCase.Params(
+                fileId = file.id!!,
```

Comment:
We should not add dose '!!' because it's a source of future crashes.

Reply:
Done